hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -455,44 +459,47 @@ public void removeTrigger(String tenantId, String triggerId) {
     }
 
     private void removeTrigger(Trigger trigger) {
+        final String tenantId = trigger.getTenantId();
+        final String triggerId = trigger.getId();
+
+        // If necessary, clean up working memory
```

Comment:
I saw that as well, but I don't think so, RHS actions complete before another rule recognition cycle.  I'll check but this is typically a fundamental aspect of the rule cycle.